### ISSUE TITLE
Support asynchronous functions and methods as keywords

### ISSUE THREAD

**Issue by KotlinIsland** [robotframework/robotframework#4089]
Say I want to have an `async def` as a keyword implementation, currently it will fail silently:
```py
from asyncio import gather

async def foo():
    log("start")
    await gather(something(1), something(2), something(3))
    log("done")
```
```robot
*** Test Cases ***
test
    foo
```
## actual result:
Nothing from the coroutine is executed and nothing appears in the log:
![image](https://user-images.githubusercontent.com/65446343/133693521-aa5f0cbe-7061-4233-82b9-3dfbebf30e3d.png)

### command line output:
```
RuntimeWarning: coroutine 'foo' was never awaited
```

## Workaround
```py
from asyncio import run, gather

def foo():
    run(_foo())


async def _foo():
    log("start")
    await gather(something(1), something(2), something(3))
    log("done")
```
Now it works, BUT if the event loop is already started by something else then this won't work :(

you would have to add some nonsense to start a second loop like:
```py
def run(coro):
    t = Thread(target=lambda: asyncio.run(coro))
    t.start()
    t.join()
```

# Request
Make async keywords await by default.

Another option would be to add an `Await Keyword` keyword, but I don't like the sound of that

**Comment by pekkaklarck** [comment]
It has never even occurred to me to use async functions as keywords, but I guess that's something we could support now that we are dropping Python 2 support in RF 5.0. How should that work in practice? Should Robot check that if a keyword is async and then automatically wait for it?

Are you interested to implement this support? Notice that it might be better to wait until Python 2 support has been removed because that's likely to change to related code a bit.

**Comment by KotlinIsland** [comment]
Maybe it could be as simple as if the result of calling the keyword is a coroutine then await it. Doesn't need to drop 2 support to do that right?

**Comment by KotlinIsland** [comment]
I'd definitely be interested in implementing it.

**Comment by pekkaklarck** [comment]
We are anyway dropping Python 2 support in RF 5.0.

**Comment by KotlinIsland** [comment]
# 😳

Could you please add python 3 support back in as It's what I use myself.

**Comment by pavlemutic** [comment]
Hey guys, any updates on this? :)

**Comment by pekkaklarck** [comment]
We have  PR #4677 implementing this and on a very quick look it looks good. I don't have time for more detailed review now, but I add this issue to RF 6.1 scope and try to have time to look at it this week.

**Comment by pekkaklarck** [comment]
This has been implemented in PR #4677 by @ygorpontelo. Thanks for a great contribution!

For some reason merging the PR failed but changes were nevertheless committed in a9160a8d53d68d121f7731a6e98e9eefc6a179da.
